pkg/portal/config/app.go

```diff
@@ -1,15 +1,76 @@
 package config
 
+import (
+	"errors"
+	"strings"
+)
+
 type AppConfig struct {
 	HostTemplate string              `envconfig:"HOST_TEMPLATE"`
 	IDPattern    string              `envconfig:"ID_PATTERN" default:"^[a-z0-9][a-z0-9-]{2,30}[a-z0-9]$"`
 	Secret       AppSecretConfig     `envconfig:"SECRET"`
 	Kubernetes   AppKubernetesConfig `envconfig:"KUBERNETES"`
 }
 
+type AppTLSCertType string
+
+const (
+	AppTLSCertNone        AppTLSCertType = "none"
+	AppTLSCertStatic      AppTLSCertType = "static"
+	AppTLSCertCertManager AppTLSCertType = "cert-manager"
+)
+
+type AppTLSCertSource struct {
+	Type AppTLSCertType
+
+	// for static type
+	SecretName string
+
+	// for cert-manager type
+	IssuerKind string
+	IssuerName string
+}
+
+func ParseTLSCertSource(desc string) (*AppTLSCertSource, error) {
+	parts := strings.Split(desc, ":")
+	switch AppTLSCertType(parts[0]) {
+	case AppTLSCertNone:
+		if len(parts) != 1 {
+			return nil, errors.New("'none' certificate type expects no arguments")
+		}
+		return &AppTLSCertSource{
+			Type: AppTLSCertNone,
+		}, nil
+
+	case AppTLSCertStatic:
+		if len(parts) != 2 {
+			return nil, errors.New("'static' certificate type expects 1 argument")
+		}
+		return &AppTLSCertSource{
+			Type:       AppTLSCertStatic,
+			SecretName: parts[1],
+		}, nil
+
+	case AppTLSCertCertManager:
+		if len(parts) != 3 {
+			return nil, errors.New("'static' certificate type expects 2 arguments")
+		}
+		return &AppTLSCertSource{
+			Type:       AppTLSCertCertManager,
+			IssuerKind: parts[1],
+			IssuerName: parts[2],
+		}, nil
+
+	default:
+		return nil, errors.New("unknown certificate type")
+	}
+}
+
 type AppKubernetesConfig struct {
-	NewResourcePrefix   string `envconfig:"NEW_RESOURCE_PREFIX" default:"app-"`
-	IngressTemplateFile string `envconfig:"INGRESS_TEMPLATE_FILE"`
+	NewResourcePrefix    string `envconfig:"NEW_RESOURCE_PREFIX" default:"app-"`
+	IngressTemplateFile  string `envconfig:"INGRESS_TEMPLATE_FILE"`
+	DefaultDomainTLSCert string `envconfig:"DEFAULT_DOMAIN_TLS_CERT" default:"none"`
+	CustomDomainTLSCert  string `envconfig:"CUSTOM_DOMAIN_TLS_CERT" default:"none"`
```

Comment:
Instead of encoding the value as colon separated, can we use a struct?

```golang
type AppKubernetesConfig struct {
  DefaultDomainTLSCert TLSCertConfig string `envconfig:"DEFAULT_DOMAIN_TLS_CERT"`
}

type TLSCertConfig struct {
  Type string `envconfig:"TYPE" default:"none"`
  
  SecretName string `envconfig:"SECRET_NAME"`
  
  IssuerKind string `envconfig:"ISSUER_KIND"`
  IssuerName string `envconfig:"ISSUER_NAME"`
}
```